src/Raven.Server/Documents/Sharding/Handlers/Admin/ShardedAdminDatabasesHandler.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.Threading.Tasks;
+using Raven.Server.Routing;
+using Raven.Server.Utils;
+using Raven.Server.Web;
+
+namespace Raven.Server.Documents.Sharding.Handlers.Admin
+{
+    public class ShardedAdminDatabasesHandler : RequestHandler
+    {
+        [RavenAction("/admin/databases/sharded/wait-for", "POST", AuthorizationStatus.Operator)]
+        public async Task WaitFor()
+        {
+            var database = GetStringQueryString("database");
+            var index = GetLongQueryString("index");
+
+            var result = Server.ServerStore.DatabasesLandlord.TryGetOrCreateDatabase(database);
+            if (result.DatabaseStatus != DatabasesLandlord.DatabaseSearchResult.Status.Sharded)
+                throw new NotSupportedException("This endpoint is only for sharded database");
+
+            await result.DatabaseContext.ShardedDatabaseContextIndexNotifications.WaitForIndexNotification(index, HttpContext.RequestAborted);
```

Comment:
This is doing the wait on a _single_ node in the cluster. Is that your intent? Who is actually going to call this?